tests/unit/lms/services/group_info_test.py

```diff
@@ -0,0 +1,124 @@
+from unittest import mock
+
+import pytest
+
+from lms.models import ApplicationInstance, GroupInfo
+from lms.services.group_info import GroupInfoService
+
+
+class TestGroupInfoUpsert:
+    AUTHORITY = "TEST_AUTHORITY_PROVIDED_ID"
+    CONSUMER_KEY = "TEST_CONSUMER_KEY"
+
+    def test_it_adds_a_new_GroupInfo_if_none_exists(
+        self, db_session, group_info_svc, params
+    ):
+        group_info_svc.upsert(
+            authority_provided_id=self.AUTHORITY,
+            consumer_key=self.CONSUMER_KEY,
+            params=params,
+        )
+
+        group_info = self._get_inserted_group_info(db_session)
+
+        assert group_info.consumer_key == self.CONSUMER_KEY
+        assert group_info.context_title == params["context_title"]
+        assert group_info.context_label == params["context_label"]
+
+    def test_it_updates_an_existing_GroupInfo_if_one_already_exists(
+        self, db_session, group_info_svc, params
+    ):
+        db_session.add(
+            GroupInfo(
+                **dict(
+                    params,
+                    id=None,
+                    authority_provided_id=self.AUTHORITY,
+                    consumer_key=self.CONSUMER_KEY,
+                )
+            )
+        )
+
+        group_info_svc.upsert(
+            authority_provided_id=self.AUTHORITY,
+            consumer_key=self.CONSUMER_KEY,
+            params=dict(params, context_title="NEW_TITLE"),
+        )
+
+        group_info = self._get_inserted_group_info(db_session)
+
+        assert group_info.consumer_key == self.CONSUMER_KEY
+        assert group_info.context_label == params["context_label"]
+        assert group_info.context_title == "NEW_TITLE"
+
+    def test_it_ignores_non_metadata_params(self, db_session, group_info_svc, params):
+        group_info_svc.upsert(
+            authority_provided_id=self.AUTHORITY,
+            consumer_key=self.CONSUMER_KEY,
+            params=dict(
+                params,
+                id="IGNORE ME 1",
+                authority_provided_id="IGNORE ME 2",
+                something_unrelated="IGNORED ME 3",
+            ),
+        )
+
+        group_info = self._get_inserted_group_info(db_session)
+
+        assert group_info.authority_provided_id == self.AUTHORITY
+
+    def _get_inserted_group_info(self, db_session):
```

Comment:
I don't think leading `_`'s are needed in test class helper methods

Reply:
Fixed